pyop2/types/dat.py

```diff
@@ -1021,6 +1021,8 @@ def vec_context(self, access):
                     size = v.local_size
                     array[offset:offset+size] = v.array_r[:]
                     offset += size
+            del array
+
```

Comment:
I would prefer to make this lifetime explicitly managed:
```
with self._vec.array as array:
    for d in self:
        with d.vec_ro as v:
             ....
yield self._vec
....
```

Reply:
Ok thanks! I think it is rather: `with self._vec as array` since `self._vec.array` is the numpy array. I am not sure if the PETSc Vec context manager will call `Vec.array` behind the scenes but we the counter does increase which is what we are after.

Reply:
ah yes, it does